src/go/rpk/pkg/config/manager.go

```diff
@@ -249,6 +249,7 @@ func (m *manager) Write(conf *Config) error {
 	// Merge the config into a new viper.Viper instance to prevent
 	// concurrent writes to the underlying config map.
 	v := InitViper(m.fs)
+	v.MergeConfigMap(m.v.AllSettings())
```

Comment:
I find it confusing that a type that contains a configuration, has a member function that takes a configuration, and the operation performed merges the configurations and writes them to disk, and that operation is called `Write`.

But I guess that can be pushed to another PR.

Reply:
Totally. I agree with you in that `Write` shouldn't require a config instance.
I liked what you suggested about splitting it into `Merge(conf *Config) error`  & `Write() error`